mlflow/utils/autologging_utils.py

```diff
@@ -187,3 +192,98 @@ def resolve_input_example_and_signature(
         logger.warning(model_signature_user_msg)
 
     return input_example if log_input_example else None, model_signature
+
+
+# wrapper functions to be able to mock this easily in the tests
+def time_wrapper_for_log():
+    return time.time()
+
+
+def time_wrapper_for_current():
+    return time.time()
+
+
+def time_wrapper_for_timestamp():
```

Comment:
Instead of wrapping `time.time()` for mocking, can we just manipulate the `total_log_batch_time` and `total_training_time` properties of `BatchMetricsHandler` in our test cases?

If we're concerned about the measurement of `total_training_time` and `total_log_batch_time`, we can always construct another test case that performs sleeps to simulate training / logging and then verifies that `total_training_time` / `total_log_batch_time` exceed expected thresholds.

Reply:
i tried to test this but for some reason at least in pytest sleep doesnt increase the system clock, even though the test is obviously taking longer (so the sleep is running)